pkg/diagnose/connectivity/core_endpoint.go

```diff
@@ -213,3 +227,25 @@ func noResponseHints(err error) string {
 
 	return ""
 }
+
+func clientWithOneRedirects() *http.Client {
+	clientCustom := &http.Client{}
+	clientCustom.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	}
+	return clientCustom
+}
+
+// See if the URL is redirected to another URL, and return the status code of the redirection
+func acceptRedirection(url string, client *http.Client) (int, error) {
+	res, err := client.Head(url)
+	if err != nil {
+		return res.StatusCode, err
+	}
+	defer res.Body.Close() // Close the response body
+	// Since the flare only accept post, when we perform a head request, we should expect a redirection before the final response
+	if res.StatusCode == http.StatusTemporaryRedirect || res.StatusCode == http.StatusPermanentRedirect || res.StatusCode == http.StatusOK {
+		return res.StatusCode, err
+	}
+	return res.StatusCode, err
```

Comment:
Huuum either there is an issue here or you can remove the `if` entirely 😄

Reply:
issue for sure this code is not good